### PR TITLE
Breaking: convert to ECMAScript Modules (ESM)

### LINKED ISSUE #1364: Convert to ECMAScript Modules (ESM)
### Overview

This package depends on the `chalk` npm package, which is now ESM-only. Now that ESM is supported in all relevant recent versions of Node I'd like to be able to update to the latest Chalk & other packages.

Yay, JS modules progress!

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## PR Checklist

-   [x] Addresses an existing issue: fixes #1364
-   [x] That issue was marked as [`status: accepting prs`](https://github.com/typescript-eslint/tslint-to-eslint-config/labels/status%3A%20accepting%20prs)

## Overview

Will fill this out soon. Roughly:

* used SWC for speed and simplicity over Babel
* chalk and lodash were a pain pending https://github.com/facebook/jest/issues/12270

Note that this will need to be released in beta for a bit as a major version breaking change.

### CODE REVIEW

File: lodash-es.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export const wat = import("./node_modules/lodash/isEqual");
+
+console.log({ wat });
```
Comment (JoshuaKGoldberg) [inline lodash-es.js:3]: (will delete file)

```suggestion
```

File: package.json [hunk 2]
```diff
@@ -22,28 +22,26 @@
         "typescript": "4.5.5"
     },
     "devDependencies": {
-        "@babel/core": "7.17.5",
-        "@babel/plugin-proposal-nullish-coalescing-operator": "7.16.7",
-        "@babel/plugin-proposal-optional-chaining": "7.16.7",
-        "@babel/preset-env": "7.16.11",
-        "@babel/preset-typescript": "7.16.7",
+        "@swc/core": "^1.2.143",
```
Comment (Josh-Cena) [inline package.json:25]: I'm also seeking to convert Docusaurus to ESM (https://github.com/facebook/docusaurus/issues/6520) and I've just been figuring out how to make Jest happy! Is this Babel -> SWC necessary, or is it just a quality-of-life improvement?

Comment (JoshuaKGoldberg) [reply package.json:25]: Just quality of life, yeah. I'll probably split these changes out to make the PR and blog post more clear.

Comment (JoshuaKGoldberg) [reply package.json:25]: https://github.com/typescript-eslint/tslint-to-eslint-config/issues/1366

**Pull request merged**